lib/bloc/account/account_model.dart

```diff
@@ -737,7 +737,10 @@ class SinglePaymentInfo implements PaymentInfo {
       return PaymentVendor.FAST_BITCOINS;
     }
 
-    if (type == PaymentType.RECEIVED && memo.payeeName != null) {
+    final payeeName = memo.payeeName;
+    if (type == PaymentType.RECEIVED &&
+        payeeName != null &&
+        payeeName.isNotEmpty) {
```

Comment:
This heuristic works indeed. We could also fetch all sale items when building the payments array and inject the sale into the payments but this might be sufficient for now.
Could you please wrap this in a getter (isSalePayment) so later when this heuristic changes we will easily replace it? Also for this code to be more easily understood?

Reply:
Of course, I'll do it.
I was avoiding reading the sale items to not hurt the initial build time of the home.

Reply:
✅